### ISSUE TITLE
General approach with async in BehaviorTestKit

### ISSUE THREAD

**Issue by johanandren** [akka/akka#24731]
I think we should take a step back and think about the general approach with async things in the BehaviorTestKit, as things are moving now it's somewhere in-between with Futures being run but spawned children being faked.

I think we should aim to make it very clear that it is not for async testing and have scheduling events/scheduling perhaps captured as `Effects` but not execute them and point people who wants to do async use the `ActorTestKit` instead.

Relevant issues sparking this: #24686, #24703

**Comment by johanandren** [comment]
Ping @raboof @chbatey @patriknw

**Comment by raboof** [comment]
Yes, that's indeed confusing right now. Perhaps it would make sense to provide an easy (but still explicit) way to trigger evaluation of captured async/scheduled calls, and link to the async `ActorTestKit` in the javadocs for those tools?

**Comment by johanandren** [comment]
If the users can get a Runnable or something out of the effect for scheduling a Future, and the details of scheduling a timed event, they can run it themselves?

**Comment by chbatey** [comment]
It would be a shame not to support timers (https://github.com/akka/akka/issues/24686) with the explicit scheduler.

**Comment by johanandren** [comment]
Related issue: #24825
